### DISCUSSION TITLE
download the demo code, but can't work well in my platform.

### DISCUSSION THREAD

**Discussion by Aline-liyang** [keycloakify/keycloakify#244] [Q&A]
Hello, i want to try this demo code in my platform.  The following is what i did:
- download the demo code.
- excute the cmd
  1) npm install
  2) npm install --global yarn
  3) PS C:\yangli\internship\practice\keycloakify-starter-main> yarn version
yarn version v1.22.19
info Current version: 1.0.9
question New version:
Done in 4.07s.
 4) yarn build
 5) yarn keycloak 

 till now, all the things look right.
- then i execute "yarn start"
×
```
Error: This app is a Keycloak themeIt isn't meant to be deployed outside of Keycloak
./src/index.tsx
http://localhost:3000/static/js/main.chunk.js:26:9
  23 | var _getKcContext = Object(keycloakify__WEBPACK_IMPORTED_MODULE_2__["getKcContext"])(),
  24 |   kcContext = _getKcContext.kcContext;
  25 | if (kcContext === undefined) {
> 26 |   throw new Error("This app is a Keycloak theme" + "It isn't meant to be deployed outside of Keycloak");
     |         ^  27 | }
  28 | Object(react_dom_client__WEBPACK_IMPORTED_MODULE_1__["createRoot"])(document.getElementById("root")).render( /*#__PURE__*/React.createElement(keycloakify__WEBPACK_IMPORTED_MODULE_2__["default"], Object.assign({
  29 |   kcContext: kcContext
![image](https://user-images.githubusercontent.com/99462135/217101295-35ade712-1065-4a4a-bf46-c3366365604e.png)
```

I didn't modify any code except "kcContext.ts", in this file, I only uncomment line 5

```

import { getKcContext } from "keycloakify/lib/getKcContext";

export const { kcContext } = getKcContext({
    /* Uncomment to test the login page with mock data */
    "mockPageId": "login.ftl",
    "mockData": [
        {
            "pageId": "login.ftl",
            "locale": {
                "currentLanguageTag": "fr", //When we test the login page we do it in french
            },
        },
    ],
});

export type KcContext = NonNullable<typeof kcContext>;

```
and in package.json, if i modified nothing, the following error will exist:
```

Error: error:0308010C:digital envelope routines::unsupported
    at new Hash (node:internal/crypto/hash:71:19)
    at Object.createHash (node:crypto:133:10)
    at module.exports (C:\yangli\internship\practice\keycloakify-starter-main\node_modules\webpack\lib\util\createHash.js:90:53)
    at NormalModule._initBuildHash (C:\yangli\internship\practice\keycloakify-starter-main\node_modules\webpack\lib\NormalModule.js:386:16)
    at C:\yangli\internship\practice\keycloakify-starter-main\node_modules\webpack\lib\NormalModule.js:418:10
    at C:\yangli\internship\practice\keycloakify-starter-main\node_modules\webpack\lib\NormalModule.js:293:13

```
so i need modify json like this: 
```
...
"scripts": {
        "start": "react-scripts --openssl-legacy-provider start",
        "build": "react-scripts --openssl-legacy-provider build",
        "keycloak": "yarn build && keycloakify",
        "download-builtin-keycloak-theme": "download-builtin-keycloak-theme 15.0.2"
    },
...

```

I deleted all the code and try 2-3 times, it seems the same issue, so i think that i must miss some steps.
could you help me to check if i missed to do some configuration? Thanks a  lot.

**Comment by garronej** [comment]
Hello @Aline-liyang,  
I'm looking into it

**Comment by garronej** [comment]
Hey @Aline-liyang,  
What version of node and Widnows are you using?  
Could you commit your changes you made and share the link of the repo?

**Comment by garronej** [comment]
Most likely is you haven't saved your changes made to kcContext before running `yarn start`

**Comment by Aline-liyang** [comment]
Hello,

Today we have another try, the following is the steps:
- in file "index.tsx", we comment line 9 : // const { kcContext } =
getKcContext();
and we added and saved it.
import { kcContext } from "./KcApp/kcContext";
import React from 'react';

- and execute the "yarn start", it seems work well
[image: image.png]

I wonder to know if it's the correct way to solve this problem?
Thanks.

Aline



Joseph Garrone ***@***.***> 于2023年2月6日周一 20:00写道：

**Comment by Aline-liyang** [comment]
Hello Joseph,

Here is the link sharing for my code, please check it, Thanks.
https://github.com/Aline-liyang/keycloackify-test



Aline Yang ***@***.***> 于2023年2月7日周二 09:45写道：

**Comment by garronej** [comment]
Hi @Aline-liyang,  
Can't reproduce but I noticed you are running a 4 years old version of `create-react-app`!!?  
Have you downgraded on purpose `react-script` from v5 to v2?  
Also I recommend using yarn instead of npm, at least for react related apps.  

I'd recommend upgrading node.js and starting fresh from https://github.com/garronej/keycloakify-advanced-starter#quick-start  
running precisely the commands in the quick start guide.  

Don't hesitate to get back to me if you still have issues.

**Comment by Aline-liyang** [comment]
Thanks Joseph, i will try the project that you recommended.


Joseph Garrone ***@***.***> 于2023年2月7日周二 10:37写道：

**Comment by Aline-liyang** [comment]
Hello Joseph,
yes, using the version which you recommend, it works well in my platform.
Thanks a lot.

Joseph Garrone ***@***.***> 于2023年2月7日周二 10:37写道：

**Comment by Aline-liyang** [comment]
hello Joseph,
I met another issue, i want to check the code/css for login page, i think
the composant "KcAppBase"  is for login page, but i can't find the where is
"KcAppBase" ?
Thanks.

Joseph Garrone ***@***.***> 于2023年2月7日周二 10:37写道：

**Comment by garronej** [comment]
KcAppBase if for the pages that you haven't modified at the HTML level.  
They are the default pages.  
It you want to modify the login at the html level you have to add "login.frl" here: https://github.com/garronej/keycloakify-advanced-starter/blob/main/src/KcApp/KcApp.tsx#L35-L39  
The reference code is here: https://github.com/InseeFrLab/keycloakify/blob/main/src/lib/components/Login.tsx
